packages/solutions/module-tools/src/builder/feature/swc.ts

```diff
@@ -102,6 +102,10 @@ export const swcTransform = (userTsconfig: ITsconfig) => ({
                 useDefineForClassFields,
                 legacyDecorator: emitDecoratorMetadata ? true : undefined,
                 decoratorMetadata: emitDecoratorMetadata ? true : undefined,
+                decoratorVersion:
+                  !emitDecoratorMetadata && tsVersion >= 5
+                    ? '2022-03'
+                    : undefined,
```

Comment:
This could be a breaking change for projects using ts^5 and `emitDecoratorMetadata` not enabled?

Reply:
AFAIK this option takes effect only when decorator grammar is used?

Reply:
- Just took a look into the codebase, you're right that this PR won't break user when `emitDecoratorMetadata` is set. 
- However, as per https://github.com/web-infra-dev/modern.js/blob/main/packages/solutions/module-tools/src/builder/feature/index.ts#L35-L39, SWC transform is enabled only in certain conditions. New trigger condition is needed to enable the `'2022-03'` config.
- `tsVersion > 5` is not sufficient, user could still use legacy decorator with `experimentalDecorators` flag, https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#differences-with-experimental-legacy-decorators.
- esbuild start to support the stage3 decorator since https://github.com/evanw/esbuild/releases/v0.21.0 with heuristic legacy backward support. IMO, bumping esbuild to latest version could be a way more straightforward solution.

Reply:
🤔 I'm fine with either. If you are willing to add another trigger option I'd like to help implement it. If you can bump to latest esbuild that'd be great 😀

Reply:
I'd prefer bumping esbuild as we should catch up the latest version and the feature is resolved by the way. We're now using `0.19.2` version of esbuild which is release 10 months ago. It's better to bump it, @chenjiahan is there any issue blocks bumping esbuild that I don't know?

Reply:
Talked with @chenjiahan, there's no blocking point stopping us from bumping esbuild. We'd inclined to bump esbuild to address this issue.
I'm closing this PR in favor of bumping esbuild. Let me know if you're interested in implement this. Thanks for the PR anyway.